python/pyspark/pandas/series.py

```diff
@@ -4548,22 +4548,31 @@ def update(self, other: "Series") -> None:
         if not isinstance(other, Series):
             raise TypeError("'other' must be a Series")
 
-        combined = combine_frames(self._psdf, other._psdf, how="leftouter")
+        if same_anchor(self, other):
+            scol = F.when(other.spark.column.isNotNull(), other.spark.column).otherwise(
+                self.spark.column
+            )
+            internal = self._psdf._internal.with_new_spark_column(
+                self._column_label, scol  # TODO: dtype?
+            )
+            self._psdf._update_internal_frame(internal.resolved_copy)
```

Comment:
I don't think we need `resolved_copy` to keep the anchor same.

It also fixes the issue below:

```py
>>> pdf = pd.DataFrame(
...     {"a": [None, 2, 3, 4, 5, 6, 7, 8, None], "b": [None, 5, None, 3, 2, 1, None, 0, 0]},
... )
>>> pser = pdf.a
>>> pser.update(pdf.b)
>>> pser
0    NaN
1    5.0
2    3.0
3    3.0
4    2.0
5    1.0
6    7.0
7    0.0
8    0.0
Name: a, dtype: float64
```

whereas:

```py
>>> psdf = ps.from_pandas(pdf)
>>> psser = psdf.a
>>> psser.update(psdf.b)
>>> psser
0    NaN
1    2.0
2    3.0
3    4.0
4    5.0
5    6.0
6    7.0
7    8.0
8    NaN
Name: a, dtype: float64
```

Reply:
Could you also modify tests to check above?

Reply:
yeah, thank you, done.